server/src/test/java/umm3601/user/UserControllerSpec.java

```diff
@@ -203,6 +211,38 @@ public void GetUsersByAge() throws IOException {
     }
   }
 
+  @Test
+  public void getUsersByAge() throws JsonMappingException, JsonProcessingException {
+    Context ctx = mock(Context.class);
+    // When the controller calls `ctx.queryParamMap`, return the expected map for an
+    // "?age=37" query.
+    when(ctx.queryParamMap()).thenReturn(Map.of(UserController.AGE_KEY, List.of("37")));
+    // When the controller calls `ctx.queryParamAsClass() to get the value associated with
+    // the "age" key, return an appropriate Validator. TBH, I never did figure out what the
+    // third argument to the Validator constructor was for, but `null` seems OK. I'm also not sure
+    // what the first argument is; it appears that you can set it to anything that isn't
+    // null and it's happy.
+    Validator<Integer> validator = new Validator<Integer>("age", 37, null);
+    when(ctx.queryParamAsClass(UserController.AGE_KEY, Integer.class)).thenReturn(validator);
```

Comment:
In another commit #https://github.com/UMM-CSci-3601/3601-iteration-template/commit/8a7328baf9fdbf4dfa3f6206e2aeae76932994ea we have something similar that creates a validator in a different way:
```typescript
when(ctx.queryParamAsClass("age", Integer.class))
      .thenReturn(Validator.create(Integer.class, "37", "age"));
```

Reply:
These are pretty much the same. I like the simpler construction of the `Validator` in the second version, but I probably like giving it a name like we do in the first version. I also definitely prefer having `UserController.AGE_KEY` instead of magic strings everywhere.